neonvm/controllers/virtualmachine_controller.go

```diff
@@ -321,6 +321,11 @@ func (r *VirtualMachineReconciler) doReconcile(ctx context.Context, virtualmachi
 	// Generate runner pod name
 	if len(virtualmachine.Status.PodName) == 0 {
 		virtualmachine.Status.PodName = names.SimpleNameGenerator.GenerateName(fmt.Sprintf("%s-", virtualmachine.Name))
+		// Update the .Status.PodName on API Server to avoid creating multiple pods for a single VM
+		// See https://github.com/neondatabase/autoscaling/issues/794 for the context
+		if err := r.Status().Update(ctx, virtualmachine); err != nil {
+			return fmt.Errorf("Failed to update .Status.PodName: %w", err)
+		}
 	}
```

Comment:
For later: When I realized the pod name was set _here_ (and not inside of the phase switch statement), I was quite surprised. We might want to move it into the handling for Pending or something.

Maybe not necessary until #733.

Reply:
I agree, let's leave it once we get to #733.